lib/mito/utilities/SegmentedContainer.h

```diff
@@ -161,17 +160,20 @@ namespace mito::utilities {
             return pointer;
         }
 
-        auto erase(const mito::utilities::shared_ptr<T, true> & element) -> void
```

Comment:
Yes it was intentional. Because erasing `element` from the segmented container implies that the pointed item is no longer valid (the item may be overwritten by another one), the shared pointer needs to be `reset` in this function. The `reset` method is clearly not `const`, hence the drop of the `const`ness.